File: presentation/src/main/java/pl/mobly/simplylauncher/common/components/modules/DomainModule.java
package pl.mobly.simplylauncher.common.components.modules;


import android.app.Application;

import dagger.Module;
import dagger.Provides;
import pl.mobly.data.SystemApiImpl;
import pl.mobly.domain.LoadInstalledAppsUseCase;
import pl.mobly.domain.SystemApi;
import pl.mobly.domain.interactors.SystemInfoInteractorImpl;

@Module
public class DomainModule {
    private Application application;

    public DomainModule(Application application) {
        this.application = application;
    }

    @Provides
    LoadInstalledAppsUseCase provideLoadInstalledAppsUseCase(SystemApiImpl systemApi) {
        return new LoadInstalledAppsUseCase(systemApi);
    }

    @Provides
    SystemInfoInteractorImpl provideSystemInfoInteractor(LoadInstalledAppsUseCase loadInstalledAppsUseCase) {
        return new SystemInfoInteractorImpl(loadInstalledAppsUseCase);
    }
}

File: presentation/src/main/java/pl/mobly/simplylauncher/common/components/modules/DomainModule.kt
package pl.mobly.simplylauncher.common.components.modules

import android.app.Application
import dagger.Module
import dagger.Provides
import pl.mobly.data.SystemApiImpl
import pl.mobly.domain.LoadInstalledAppsUseCase
import pl.mobly.domain.interactors.SystemInfoInteractorImpl


@Module class DomainModule(private val application: Application) {

    @Module class DomainModule(application: Application)

    @Provides
    fun provideLoadInstalledAppsUseCase(systemApiImpl: SystemApiImpl): LoadInstalledAppsUseCase {
        return LoadInstalledAppsUseCase(systemApiImpl)
    }

    @Provides
    fun provideSystemIfoInteractorImpl(loadInstalledAppsUseCase: LoadInstalledAppsUseCase): SystemInfoInteractorImpl {
        return SystemInfoInteractorImpl(loadInstalledAppsUseCase)
    }
}
